### PR TITLE
Add test cases for sub-scenario 5.1.1 & 5.1.2

### PR DESCRIPTION
**Pull request by RidmiR**
## Purpose
This PR including the test cases for sub scenario 5.1.1 and 5.1.2 with the synapse configuration app. Covered test cases are 5.1.1.1.1, 5.1.1.1.2, 5.1.2.1.1

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package org.wso2.carbon.esb.scenario.test;
```
Comment (dulanjalidilmi) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java]: Shall we add the license header?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/pom.xml [hunk 2]
```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (dulanjalidilmi) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/pom.xml]: Shall we add correct license header? Year should be 2019. Shall we update other files as well?

File: product-scenarios/5-route-messages-between-systems/pom.xml [hunk 3]
```diff
@@ -35,6 +35,7 @@
     <modules>
         <module>05-synapseConfigProject</module>
         <module>5.3-load-balance-messages-among-systems</module>
+        <module>5.1-Route-based-on-the-content-of-the-messages</module>
```
Comment (dulanjalidilmi) [inline product-scenarios/5-route-messages-between-systems/pom.xml]: Shall we rearrange the order of modules according to scenario number?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.esb.scenario.test.common.http.HttpConstants;
+
+/**
+ * This test class test routing messages based on the regular expression with valid, Invalid, Null or empty cases
+ */
+
```
[the remark this answers is not in the document]
Comment (RidmiR) [reply product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java]: Don't we need an overall description of the test class?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java (hunk 1 above)
Comment (RidmiR) [reply product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java]: Done

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/pom.xml (hunk 2 above)
Comment (RidmiR) [reply product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/pom.xml]: Done

File: product-scenarios/5-route-messages-between-systems/pom.xml (hunk 3 above)
Comment (RidmiR) [reply product-scenarios/5-route-messages-between-systems/pom.xml]: Done

File: product-scenarios/5-route-messages-between-systems/05-synapseConfigProject/05-synapseConfig/src/main/synapse-config/endpoints/05_EP_XML.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/05-synapseConfigProject/05-synapseConfig/src/main/synapse-config/endpoints/05_EP_XML.xml]: Shall we add copyrights to all the files in the synapseConfig module?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/pom.xml [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.build.directory}/jacoco</outputDirectory>
+                            <includeTypes>jar</includeTypes>
+                            <includeArtifactIds>org.jacoco.agent</includeArtifactIds>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/pom.xml]: Do we need these plugins?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/pom.xml [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        </plugins>
+    </build>
+    <dependencies>
+        <dependency>
+            <groupId>org.wso2.ei</groupId>
+            <artifactId>admin-clients</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.wso2.carbon.mediation</groupId>
+                    <artifactId>org.wso2.carbon.rest.api</artifactId>
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/pom.xml]: Do we need these dependencies?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+    }
+
+    @Test(description = "5.1.1.1.1")
+    public void routeMessagesBasedOnXpathWithValidCaseName() throws Exception {
+        String header = "5_1";
+        String url = getApiInvocationURLHttp("5_1_1_Routing_messages_based_on_content_of_message_test");
+
+        String request ="<m:GetStockPrice xmlns:m=\"http://www.example.org/stock\">\n" +
+                "   <m:StockName>IBM</m:StockName>\n" +
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java]: Line wrapping needs to happen before an operator. Shall we correct in all places?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+ * This test class test routing messages based on the Xpath with valid, Invalid, Null or empty cases
+ */
+public class RoutingBasedOnXpathTest extends ScenarioTestBase {
+
+    public static final Log log = LogFactory.getLog(RoutingBasedOnXpathTest.class);
+
+    @BeforeClass
+    public void init() throws Exception {
+        super.init();
+
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java]: Unnecessary new line

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.esb.scenario.test.common.http.HttpConstants;
+
+
+/**
+ * This test class test routing messages based on the Xpath with valid, Invalid, Null or empty cases
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java]: Shall we make the class comment more meaningful? May be, "This test class tests routing of messages based on valid, invalid, null and empty XPath expressions."? Also, let's rename the methods accordingly.

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        HTTPUtils.invokePoxEndpointAndAssert(url, request, HttpConstants.MEDIA_TYPE_TEXT_XML, header, expectedResponse,
+                200, "Switch messages based on given Xpath with valid case name");
+    }
+
+    @Test(description = "5.1.1.1.2")
+    public void routeMessagesBasedOnXpathWithInvalidCaseName() throws Exception {
+        String header = "5_1";
+        String url = getApiInvocationURLHttp("5_1_1_1_2_Routing_messages_based_on_Invalid_xpath_test");
+
+        String request ="<m:GetStockPrice xmlns:m=\"http://www.example.org/stock\">\n" +
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java]: We are using the same request over and over again, right? If this is a basic xml shall we define a constant at ScenarioConstants so that it can be used across test cases?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/resources/log4j.properties [hunk 12]
```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/resources/log4j.properties]: Shall we correct the year of the copyright?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+public class RoutingBasedOnRegularExpressionTest extends ScenarioTestBase {
+
+    public static final Log log = LogFactory.getLog(RoutingBasedOnRegularExpressionTest.class);
+
+    @BeforeClass
+    public void init() throws Exception {
+        super.init();
+
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java]: Let's not have unnecessary new lines.

File: product-scenarios/5-route-messages-between-systems/05-synapseConfigProject/05-synapseConfig/src/main/synapse-config/api/5_1_1_1_2_Routing_messages_based_on_Invalid_xpath_test.xml [hunk 14]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<api context="/5_1_1_1_2_Routing_messages_based_on_Invalid_xpath_test" name="5_1_1_1_2_Routing_messages_based_on_Invalid_xpath_test" xmlns="http://ws.apache.org/ns/synapse">
```
Comment (milindaperera) [inline product-scenarios/5-route-messages-between-systems/05-synapseConfigProject/05-synapseConfig/src/main/synapse-config/api/5_1_1_1_2_Routing_messages_based_on_Invalid_xpath_test.xml]: Lets follow the convention for naming APIs : [ID]_API_[Name]

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.esb.scenario.test.common.http.HttpConstants;
+
+/**
+ * This test class tests routing the messages into relevant backends based on the regular expressions by using switch mediator and filter mediator.
```
Comment (dulanjalidilmi) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java]: Shall we limit the maximum length of one line to 120?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (dulanjalidilmi) [reply product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java]: yes. lets add.

**Comment by RidmiR** [comment]
@milindaperera Added new test cases for sub scenario 5.1.1.3 and 5.1.2.1. Please review those changes in the latest commit.

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+
+    public static final Log log = LogFactory.getLog(RoutingBasedOnXpathTest.class);
+
+    @BeforeClass
+    public void init() throws Exception {
+        super.init();
+    }
+
+    @Test(description = "5.1.1.1.1")
+    public void routeMessagesBasedOnXpathWithValidCaseName() throws Exception {
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java]: Do we intend to mean "routeMessagesBasedOnValidXPath"? If so, shall we rename it that way?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+                200, "Switch messages based on given Xpath with valid case name");
+    }
+
+    @Test(description = "5.1.1.1.2")
+    public void routeMessagesBasedOnXpathWithInvalidCaseName() throws Exception {
+        String header = "5_1";
+        String url = getApiInvocationURLHttp("5_1_API_Routing_messages_based_on_content_of_message_test/" +
+                "Invalid_xpath_test_with_switchM");
+
+        String request = "<m:GetStockPrice xmlns:m=\"http://www.example.org/stock\">\n" +
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java]: It seems like the request is common for both test cases. Shall we define a String constant for this? Let's add it to scenario constants If it is a simple message that can be used by other test cases as well.

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/resources/source_files/5.1.1.1.1/5.1.1.1.1/request/5.1.1.1.1.xml [hunk 18]
```diff
@@ -0,0 +1,3 @@
+<m:GetStockPrice xmlns:m="http://www.example.org/stock">
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/resources/source_files/5.1.1.1.1/5.1.1.1.1/request/5.1.1.1.1.xml]: Seems like these files are nt being used. Could you double check?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.3-Using-filter-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathWithFilterMTest.java [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+    public void init() throws Exception {
+        super.init();
+    }
+
+    @Test(description = "5.1.1.3.1")
+    public void testRouteBasedOnXpathWithValidCaseName() throws Exception {
+        String header = "5_1";
+        String url = getApiInvocationURLHttp("5_1_API_Routing_messages_based_on_content_of_message_test" +
+                "/valid_xpath_test_with_filterM");
+        String request = "<m:GetStockPrice xmlns:m=\"http://www.example.org/stock\">\n"
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.3-Using-filter-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathWithFilterMTest.java]: IMO, we can move the message to ScenarioConstants so that it can be used everywhere. WDYT?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.3-Using-filter-mediator-on-given-xpath/src/test/resources/source_files/5.1.1.1.1/request/5.1.1.1.1.xml [hunk 20]
```diff
@@ -0,0 +1,3 @@
+<m:GetStockPrice xmlns:m="http://www.example.org/stock">
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.3-Using-filter-mediator-on-given-xpath/src/test/resources/source_files/5.1.1.1.1/request/5.1.1.1.1.xml]: Shall we remove the files, if they are not used?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/readme.md [hunk 21]
```diff
@@ -19,34 +19,60 @@ Given below is the syntax of how regular expression can be add to the switch med
...
+         <property xmlns:m="http://www.example.org/stock" name="StockName" expression="//m:GetStockPrice/m:StockName" scope="default" type="STRING"/>
+         <switch source="get-property('StockName')">
+            <case regex="IBM">
+               <call>
+                  <endpoint key="http://ei-backend.scenarios.wso2.org:9090/eiTests/XMLEndpoint"/>
+               </call>
+            </case>
+            <default/>
+         </switch>
+         <respond description="PASS"/>
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/readme.md]: Do we need to add a description? Even if the payload doesn't match the regex, the message flow will hit here.

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (sdkottegoda) [reply product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnRegularExpressionTest.java]: I suppose, @dulanjalidilmi was referring to the empty new line.

File: product-scenarios/5-route-messages-between-systems/pom.xml [hunk 22]
```diff
@@ -34,7 +34,8 @@
 
     <modules>
         <module>05-synapseConfigProject</module>
-        <module>5.3-load-balance-messages-among-systems</module>
+        <module>5.1-Route-based-on-the-content-of-the-messages</module>
+        <!--<module>5.3-load-balance-messages-among-systems</module>-->
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/pom.xml]: Why are we commenting out module 5.3?

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/resources/source_files/5.1.1.1.1/5.1.1.1.1/request/5.1.1.1.1.xml (hunk 18 above)
Comment (RidmiR) [reply product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/resources/source_files/5.1.1.1.1/5.1.1.1.1/request/5.1.1.1.1.xml]: removed unnecessary files.

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java (hunk 17 above)
Comment (RidmiR) [reply product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java]: Added String constant.

File: product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.3-Using-filter-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathWithFilterMTest.java [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+                200, "Route messages based on given Xpath with valid case name");
+    }
+
+    @Test(description = "5.1.1.3.2")
+    public void routeBasedOnInvalidXpathWithFilterM() throws Exception {
+        String header = "5_1";
+        String url = getApiInvocationURLHttp("5_1_API_Routing_messages_based_on_content_of_message_test" +
+                "/Invalid_xpath_test_with_filterM");
+        String request = ScenarioConstants.COMMON_ROUTING_REQUEST;
+        String expectedResponse = "<m:GetStockPriceResponse xmlns:m=\"http://www.example.org/stock\">\n"
```
Comment (sdkottegoda) [inline product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.3-Using-filter-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathWithFilterMTest.java]: We can extract out the response to ScenarioConstants as well.

File: product-scenarios/pom.xml [hunk 24]
```diff
@@ -75,13 +75,13 @@
     <modules>
         <module>scenarios-commons</module>
         <module>1-integrating-systems-that-communicate-in-heterogeneous-message-formats</module>
-        <!--<module>2-Bridging-systems-that-communicate-in-different-protocols</module>-->
+        <module>2-Bridging-systems-that-communicate-in-different-protocols</module>
```
Comment (sdkottegoda) [inline product-scenarios/pom.xml]: Should we uncomment these? I don't see any changes on those modules.
